agent/worker.go

```diff
@@ -278,8 +278,14 @@ func reconcileTaskState(ctx context.Context, w *worker, assignments []*api.Assig
 
 	removeTaskAssignment := func(taskID string) error {
 		ctx := log.WithLogger(ctx, log.G(ctx).WithField("task.id", taskID))
-		if err := SetTaskAssignment(tx, taskID, false); err != nil {
-			log.G(ctx).WithError(err).Error("error setting task assignment in database")
+		// if a task is no longer assigned, then we do not have to keep track
+		// of it. a task will only be unassigned when it is deleted on the
+		// manager. insteaad of SetTaskAssginment to true, we'll just remove
+		// the task now.
+		if err := DeleteTask(tx, taskID); err != nil {
+			log.G(ctx).WithError(err).Errorf(
+				"error removing de-assigned task %v", taskID,
```

Comment:
looks like the logger that's used already has `Withfield("task.id", ...)`, so this can be a plain `Error()`; no need to add the task ID

Reply:
yeah, good call, thanks.